### PR TITLE
feat(mcp): exhaustive get_purchase_order + canonical-name labels (#346)

### PR DESCRIPTION
**Pull request by dougborg**
## Summary

Second slice of issue #346 (exhaustive single-object `get_*` detail tools) — follows the pattern set by #352 (`get_customer`).

- **`get_purchase_order`** now surfaces every field Katana exposes on `RegularPurchaseOrder` (was 8/20), every `PurchaseOrderRow` field (UOM conversion, landed_cost, batch_transactions, etc.), full `additional_cost_rows` (fetched via `/po_additional_cost_rows?group_id=<po.default_group_id>`) and full `accounting_metadata` (fetched via `/purchase_order_accounting_metadata?purchase_order_id=<id>`). Two extra HTTP calls on top of the PO fetch.
- **`verify_order_document`** now embeds the full `purchase_order` in the same exhaustive shape — so callers can trace every compared value back to a concrete PO field without a follow-up lookup.
- **Canonical-name markdown** — both tools now label every field with its Pydantic name (`**status**:`, `**purchase_order_rows** (N):`, `**accounting_metadata**: []`). Empty lists render with explicit `: []` syntax instead of dangling section headers. Motivation: #346 follow-on ambiguity fix (LLM mistook a prettified header for a field value).

Scope boundaries: Prefab UI builders untouched; `list_purchase_orders` stays summary-shaped; `help.py` entries updated (append-only).

## Test plan

- [x] `test_get_purchase_order_full_field_coverage` — asserts every previously-dropped PO/row field surfaces
- [x] `test_get_purchase_order_fetches_additional_costs_and_accounting_metadata` — patches `_fetch_po_additional_cost_rows` and `_fetch_po_accounting_metadata`, asserts right identifiers + values flow through
- [x] `test_get_purchase_order_markdown_uses_canonical_field_names` — pins the canonical-name convention
- [x] `uv run poe check` — clean (2391 passed, 3 skipped)
- [x] Shared-mock updates so every existing `get_purchase_order` / `verify_order_document` test survives the expanded response shape (no test-logic regressions)

Refs: #346, #352.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 1]
```diff
@@ -487,7 +487,577 @@ async def receive_purchase_order(
...
+        if not orders:
+            raise ValueError(f"Purchase order '{request.order_no}' not found")
+        po = orders[0]
+
+    default_group_id = unwrap_unset(po.default_group_id, None)
+    additional_cost_rows = await _fetch_po_additional_cost_rows(
+        services, default_group_id
+    )
+    accounting_metadata = await _fetch_po_accounting_metadata(services, po.id)
+
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py:923]: Addressed in 2a6d5c4f — `_fetch_po_additional_cost_rows` and `_fetch_po_accounting_metadata` now run concurrently via `asyncio.gather` (applies to both `_get_purchase_order_impl` and `_verify_order_document_impl`), plus a new test asserts `gather` is invoked with two awaitables.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 2]
```diff
@@ -637,6 +1259,20 @@ async def _verify_order_document_impl(
...
         po = unwrap_as(po_response, RegularPurchaseOrder)
 
+        # Build the exhaustive PO structured view — same shape as
+        # get_purchase_order — so callers have full context on what was
+        # compared against. Side-data fetches happen on-demand.
+        default_group_id = unwrap_unset(po.default_group_id, None)
+        additional_cost_rows = await _fetch_po_additional_cost_rows(
+            services, default_group_id
+        )
+        accounting_metadata = await _fetch_po_accounting_metadata(services, po.id)
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py]: Addressed in 2a6d5c4f — same `asyncio.gather` fix applied here. The two fetches now run concurrently before `_build_get_purchase_order_response`.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 3]
```diff
@@ -487,7 +487,577 @@ async def receive_purchase_order(
...
+    created_at: str | None = None
+    updated_at: str | None = None
+    deleted_at: str | None = None
+    status: str | None = None
+    order_no: str | None = None
+    entity_type: str | None = None
+    default_group_id: int | None = None
+    supplier_id: int | None = None
+    currency: str | None = None
+    expected_arrival_date: str | None = None
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py:616]: Addressed in 2a6d5c4f — `GetPurchaseOrderResponse` now exposes a `supplier: SupplierInfo | None` field. `SupplierInfo` carries every `Supplier` field (name, email, phone, currency, comment, default_address_id, addresses, timestamps), populated in `_build_get_purchase_order_response` via a new `_supplier_info` helper. Rendered inline under `**supplier**:` with canonical labels, `**supplier**: null` when not embedded. New `test_get_purchase_order_surfaces_embedded_supplier` asserts every field flows through.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 4]
```diff
@@ -487,7 +487,577 @@ async def receive_purchase_order(
...
+        get_purchase_order as api_get_purchase_order,
+    )
+    from katana_public_api_client.models import ErrorResponse
+    from katana_public_api_client.utils import unwrap_data
+
+    if not request.order_no and not request.order_id:
+        raise ValueError("Either order_no or order_id must be provided")
+
+    services = get_services(context)
+
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py]: Addressed in 2a6d5c4f — switched to explicit `is None` checks for both `order_id` and `order_no`, with a separate guard that rejects empty-string `order_no` up front. Two new tests cover the corner cases: `test_get_purchase_order_accepts_zero_order_id_via_is_none_check` (`order_id=0` routes to get-by-id) and `test_get_purchase_order_rejects_empty_order_no` (`order_no=''` raises).

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 5]
```diff
@@ -546,60 +1116,112 @@ class VerifyOrderDocumentRequest(BaseModel):
...
+    # Embed the exhaustive PO under its canonical key so the LLM can trace
+    # every compared value back to a concrete PO field.
+    if response.purchase_order is not None:
+        md_lines.append("")
+        md_lines.append("**purchase_order**:")
+        # Indent the rendered PO block by two spaces so it nests under the
+        # purchase_order label.
+        po_md = _render_get_purchase_order_md(response.purchase_order)
+        for line in po_md.splitlines():
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py]: Addressed in 2a6d5c4f — `_render_get_purchase_order_md` now takes an `embed: bool` keyword (default False). When the verify response embeds the PO it calls `_render_get_purchase_order_md(response.purchase_order, embed=True)`, which omits the `## PO …` heading. New `test_verify_order_document_embeds_po_without_nested_heading` asserts `## PO` never appears anywhere in the verify-mode output.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 6]
```diff
@@ -546,60 +1116,112 @@ class VerifyOrderDocumentRequest(BaseModel):
...
 
 class VerifyOrderDocumentResponse(BaseModel):
-    """Response from verifying an order document."""
+    """Response from verifying an order document.
+
+    Includes the full ``purchase_order`` in the same exhaustive shape as
+    ``get_purchase_order`` so callers don't need a follow-up lookup to see
+    what was compared against. ``purchase_order`` is None only when the PO
+    fetch itself raised (the exception propagates; this field documents
+    the happy-path shape).
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py]: Addressed in 2a6d5c4f — rewrote the `VerifyOrderDocumentResponse.purchase_order` docstring. It now explains that the field is typed `Optional` only because it's assigned during `_verify_order_document_impl` rather than at model construction; in practice, consumers only observe successful responses (errors on the PO fetch propagate as exceptions), so `purchase_order` is present on every response a caller receives.

**Review by dougborg — COMMENTED** [review]
## /review-pr (Mode A — fresh code-reviewer pass)

### 🚫 BLOCKING (fixed in baa2fc0a)

1. **`_fetch_po_additional_cost_rows` and `_fetch_po_accounting_metadata` had no transport-error guard.** `unwrap_data(default=[])` only handles parsed non-200 responses; it doesn't catch `httpx.HTTPError` (timeouts, connection errors) or `UnexpectedStatus` raised before the response object is built. A transient 5xx on the side-data endpoint would crash `get_purchase_order` even when the core PO fetched fine.

   **Fixed**: wrapped each `asyncio_detailed(...)` call in `try/except (httpx.HTTPError, UnexpectedStatus)` returning `[]`, and also added `raise_on_error=False` to `unwrap_data` for symmetric handling of parsed-error responses. This matches the pattern that landed in #352 after the same Copilot finding there.

### ⚠️ SUGGESTION — declined, with rationale

- **`tracking_location_id: int | None` is too wide**: actually correct as-is. `PurchaseOrder` is a union of `RegularPurchaseOrder` (where `tracking_location_id: None | Unset`) and `OutsourcedPurchaseOrder` (where it's a required `int`). The current type covers both branches. Flagging only `None` would lose the outsourced case.

### 💬 NITPICK / follow-ups (not blocking)

- `_iso_optional` accepts any `str` without validation. The docstring says `datetime | str | None`; the type annotation should match. Low impact — deferring to a sweep across the other #346 PRs which all have the same helper shape.
- Three PO mock builders (`create_mock_po`, `_make_mock_po`, `_make_exhaustive_mock_po`) have duplicated UNSET-default field lists. Drift trap. Could extract one shared helper. Not a regression, just tech debt.
- Test assertions use positional `args[1]` — prefer `kwargs['group_id']` or `assert_awaited_once_with`.

### ✨ What looks good

- Field coverage: every field on `RegularPurchaseOrder` / `OutsourcedPurchaseOrder` / `PurchaseOrderRow` / `PurchaseOrderAdditionalCostRow` / `PurchaseOrderAccountingMetadata` / `Supplier` maps through to the corresponding Pydantic sub-model. Nothing dropped.
- `asyncio.gather` correctly applied in both `_get_purchase_order_impl` and the `verify_order_document` path.
- `embed=True` suppresses the `## PO …` heading in verify output cleanly.
- `autouse` fixture defaulting side-data fetches to `[]` prevents pollution without touching every existing test.

### Summary

Verdict: fixes pushed for the blocker, non-blocking items deferred.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 7]
```diff
@@ -487,7 +488,692 @@ async def receive_purchase_order(
...
+            continue
+        val = getattr(row, fname, None)
+        if val is None or val == "":
+            continue
+        lines.append(f"    **{fname}**: {val}")
+    if row.batch_transactions:
+        lines.append(
+            f"    **batch_transactions** ({len(row.batch_transactions)}): "
+            f"{row.batch_transactions}"
+        )
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py:1025]: Fixed — `_po_row_info_lines` now emits `**batch_transactions**: []` in the empty-list case, matching the convention.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 8]
```diff
@@ -487,7 +488,692 @@ async def receive_purchase_order(
...
+    lines: list[str] = []
+    for fname in _SUPPLIER_FIELDS:
+        val = getattr(supplier, fname, None)
+        if val is None or val == "":
+            continue
+        lines.append(f"  **{fname}**: {val}")
+    if supplier.addresses:
+        lines.append(
+            f"  **addresses** ({len(supplier.addresses)}): {supplier.addresses}"
+        )
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py:1088]: Fixed — `_render_supplier_md` now emits `**addresses**: []` when the supplier has no registered addresses.

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 9]
```diff
@@ -487,7 +488,692 @@ async def receive_purchase_order(
...
+    """Return an ISO-8601 string for a datetime-or-str value, else None.
+
+    Accepts either a ``datetime`` (calls ``.isoformat()``) or an already-
+    formatted string. Returns None when the value is None/UNSET.
+    """
+    if value is None:
+        return None
+    if hasattr(value, "isoformat"):
+        return value.isoformat()
+    return str(value)
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py]: Fixed — typed the parameter as `datetime | str | None` and switched to `isinstance(value, datetime)` rather than `hasattr("isoformat")`. Pyright will now catch callers that pass the wrong type. Also tightened the docstring to make clear the helper expects pre-unwrapped input (no UNSET sentinel).

File: katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py [hunk 10]
```diff
@@ -487,7 +488,692 @@ async def receive_purchase_order(
...
+
+    Two extra HTTP calls are made on top of the PO fetch — one for
+    additional cost rows, one for accounting metadata — so callers don't
+    need follow-up lookups for standard PO data. Use this tool whenever
+    full detail is needed; use ``list_purchase_orders`` for discovery.
+
+    Provide either order_no (e.g., 'PO-1022') or order_id.
+    """
+    from katana_mcp.tools.tool_result_utils import make_simple_result
+
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/foundation/purchase_orders.py]: Fixed — removed the redundant inner import. `make_simple_result` was already at module scope from the top-level imports block.
